Assets/Scripts/Interactables/Dialogue/NPCDialogue/NPCInteractable.cs

```diff
@@ -76,59 +76,38 @@ protected override void Start()
 
     protected override void Update()
     {
-        if(IsWithinRadius(originPosition, GameManager.Instance.GetPlayerTransform(), boundaryRadius))
+        if (IsWithinRadius(originPosition, GameManager.Instance.GetPlayerTransform(), boundaryRadius))
         {
             // Prevent loading during a conversation
-            if (!DialogueManager.Instance.CheckIsConversing() && !DialogueManager.Instance.CheckIsAutoPlaying())
+            if (!DialogueManager.Instance.IsActiveConversation(conversation))
             {
                 LoadConversation();
-            }
-
-            if (conversation.shouldFollow)
-            {
-                TriggerFollow(player);
-            }
-
 
-            // Autoplay
-            if (conversation.autoplayConversation)
-            {
-                if(!DialogueManager.Instance.CheckIsAutoPlaying())
+                // Autoplay
+                if (conversation.autoplayConversation)
                 {
                     OnInteract();
                 }
-            }
-            // Enter collider to interact
-            else
-            {
-                if(conversation.autoInitiate && !DialogueManager.Instance.CheckIsConversing())
-				{
-                    interactableOn = true;
-                    OnInteract();
-				}
-            }
-        }
-        else
-        {
-            if (isFollowing)
-            {
-                StopFollow();
+                else
+                {
+                    //Prevent from auto-initiating in the middle of advancing a conversation
+                    if (conversation.autoInitiate && !DialogueManager.Instance.CheckIsAdvancing())
+                    {
+                        //interactableOn = true;
+                        OnInteract();
+                    }
+                }
             }
         }
 
-        if (!DialogueManager.Instance.CheckIsConversing() && !DialogueManager.Instance.CheckIsAutoPlaying())
+        // TODO: Boundary Radius should always be >= Interact Radius Possible issues will arise otherwise
```

Comment:
would be fancy to use this validator actually to make sure this is always the case:
https://github.com/dbrizov/NaughtyAttributes#validateinput

Reply:
Oouuuuuuu thats dope!